socket.go

```diff
@@ -384,6 +384,9 @@ func (socket *mongoSocket) kill(err error, abend bool) {
 	socket.dead = err
 	socket.conn.Close()
 	stats.socketsAlive(-1)
+	if abend {
+		stats.failedConn(1)
```

Comment:
Any reason not to do this at the end of the function in the existing abend check?

Reply:
Mostly because I thought it would make more sense to increment this together with the `socketsAlive` metric. I don't think there's any technical reason it has to be here.